ion/timestamp.go

```diff
@@ -515,12 +518,13 @@ func (ts Timestamp) Equal(ts1 Timestamp) bool {
 		ts.numFractionalSeconds == ts1.numFractionalSeconds
 }
 
-// TruncatedNanoseconds returns nanoseconds with trailing zeros removed (ie. 123456000 gets truncated to 123456).
+// TruncatedNanoseconds returns nanoseconds with trailing values removed up to the difference of max fractional precision - time stamp's fractional precision
+// e.g. 123456000 with fractional precision: 3 will get truncated to 123.
 func (ts Timestamp) TruncatedNanoseconds() int {
```

Comment:
This function is only used internally to return the correct coefficient given the precision and is then used to write the timestamp in binary.

For example:
* `123456000` nanoseconds
* `3` fractional precision

This function will return coefficient: `123`. We will then write in binary, exponent: `3` and coefficient: `123`

So it will become `123 * 10 ^-3 = 0.123`

vs

Not truncating the nanoseconds which will be:
`123456789 * 10 ^-3 = 123456.789`